app/src/main/java/com/example/tasky/agenda/presentation/agenda_detail/AgendaDetailViewModel.kt

```diff
@@ -394,7 +430,41 @@ class AgendaDetailViewModel(
         }
     }
 
-    private fun deleteAgendaItem(id: String) {
+    private fun saveReminder() {
+        viewModelScope.launch(Dispatchers.IO) {
+            val currentTimestamp = ZonedDateTime.now(ZoneId.of("UTC"))
```

Comment:
I see you’re grabbing ZonedDateTime.now(ZoneId.of("UTC")) -  which works fine for persisting because the server expects UTC. But for the user-facing side (and for testing), you’ll almost always want to show times in the device’s local zone. The smoother approach is: device zone for picking/displaying, convert to UTC when saving, and convert back when reading.

This tripped me up too, and [Philipp’s video](https://www.youtube.com/watch?v=gzHy6wKAJh8&t) on handling time zones in Android is super helpful here  -  worth a watch to avoid those hidden “why is my reminder at 3 AM?” bugs 😅.